identity/src/commonMain/kotlin/com/android/identity/nfc/NdefRecord.kt

```diff
@@ -87,19 +91,21 @@ data class NdefRecord(
         if (isShortRecord) flags = flags or FLAG_SR
         if (idLengthPresent) flags = flags or FLAG_IL
 
-        buf.writeByte(flags.toByte())
-        buf.writeByte(type.size.toByte())
+        bsb.appendUInt8(flags)
+        bsb.appendUInt8(type.size)
         if (isShortRecord) {
-            buf.writeByte(payload.size.toByte())
+            bsb.appendUInt8(payload.size)
         } else {
-            buf.writeInt(payload.size)
+            bsb.appendUInt32(payload.size) // TODO: b/393388370 - is it really 4 byte? or perhaps it's 2 (UInt16)?
```

Comment:
It's always 4 bytes, according to the NDEF spec from NFC Forum (I have all the specs, needed them to implement this).